src/main/java/org/dataloader/impl/DefaultCacheMap.java

```diff
@@ -60,6 +61,14 @@ public CompletableFuture<V> get(K key) {
         return cache.get(key);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Collection<CompletableFuture<V>> getAll() {
+        return cache.values();
```

Comment:
Rather than expose the `getAll` down in the helper and then in the data loader,

I think we should expose the `CacheMap` itself on DataLoader

```
    private final CacheMap<Object, V> futureCache;
````

is inside DataLoader.

Change this to

```
    public  CacheMap<Object, V> getCacheMap() 
```

And for symmetry (since we are going to do this) we should expose the other atts

```
    private final StatisticsCollector stats;
    private final CacheMap<Object, V> futureCache;
    private final ValueCache<K, V> valueCache;
```

Reply:
Should we worry about making `futureCache` and `valueCache` public ? i was trying to just expose the `CacheMap` values as readonly to avoid the chance of their state.

Reply:
also was wondering, given that we are adding a new method in the `CacheMap` interface, should we define a default behavior to return maybe empty collection or null ? otherwise this could be a breaking change if ppl is using a custom implementation of `CacheMap`

Reply:
I get your point on trying to reduce danger but the fact is the DataLoaderOptions passed in the CachMap / ValueMap or it get defaulted.

I guess we could make a UnmodifiableCacheMap / ValueCache wrapper say so gain write safety


> Should we worry about making futureCache and valueCache public ?

My view is sure it could be a foot gun... so dont shoot yourself with it.   You needed access to the future cache so lets give it out.  After all it was nominally passed in.

But UnmodifiableCacheMap is not a crazy idea

Reply:
added the getters for `cacheMap` and `valueCache`, 
will think about the wrapper interface to remove the mutable methods `clear`, `delete` and `set` and probably open a separate PR, if we decide that we need it